spring-web/src/main/java/org/springframework/web/util/UriBuilder.java

```diff
@@ -185,6 +186,15 @@
 	 */
 	UriBuilder queryParam(String name, Object... values);
 
+	/**
+	 * Append the given query parameter and value if and only if a value is present in the Optional.
+	 * No action will be taken, and the query parameter name will not be added, if the passed Optional is empty.
+	 * @param name the query parameter name
+	 * @param optionalValue an Optional, either empty or holding the query parameter value.
+	 * @return
+	 */
+	UriBuilder optionalQueryParam(String name, Optional<?> optionalValue);
```

Comment:
`queryParamIfPresent` aligns better with terminology from `Optional` and it sorts better alphabetically with the existing `queryParam` methods.

Reply:
Done.